### PR TITLE
Remove extra await/state machine in FlushAsync

### PR DESCRIPTION
**Pull request by jkotalik**
Small improvement to FlushAsync for WriteOnlyPipeStream.

### CODE REVIEW

File: src/Http/Http/src/WriteOnlyPipeStream.cs [hunk 1]
```diff
@@ -81,9 +81,17 @@ public override void Flush()
         }
 
         /// <inheritdoc />
-        public override async Task FlushAsync(CancellationToken cancellationToken)
+        public override Task FlushAsync(CancellationToken cancellationToken)
         {
-            await InnerPipeWriter.FlushAsync(cancellationToken);
+            var task = InnerPipeWriter.FlushAsync(cancellationToken);
```
Comment (benaadams) [inline src/Http/Http/src/WriteOnlyPipeStream.cs]: Can this use the `.GetAsTask()` extension instead?

Comment (JamesNK) [reply src/Http/Http/src/WriteOnlyPipeStream.cs]: Place in a shared file in https://github.com/aspnet/AspNetCore/tree/master/src/Shared and reuse here, servers, and SignalR

**Review by JamesNK — CHANGES_REQUESTED** [review]

File: src/Http/Http/src/WriteOnlyPipeStream.cs (hunk 1 above)
Comment (jkotalik) [reply src/Http/Http/src/WriteOnlyPipeStream.cs]: Yeah I think it deserves to be shared now. I will once I’m not OOF.

**Review by benaadams — APPROVED** [review]

**Comment by davidfowl** [comment]
Why aren’t these files linked instead of included separately?

**Comment by jkotalik** [comment]
> Why aren’t these files linked instead of included separately

Clarify?

**Review by JamesNK — APPROVED** [review]

**Review by halter73 — APPROVED** [review]

**Comment by jkotalik** [comment]
All failures have already been filed.

**Pull request merged**